### PR TITLE
Add STS SemRel2024 (14 languages)

### PR DESCRIPTION
**Pull request by SaitejaUtpala**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [ ] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by KennethEnevoldsen** [comment]
Hi @SaitejaUtpala, this seems like it is a WIP. Will you complete the checklist - feel free to @ me afterwards

**Comment by SaitejaUtpala** [comment]
> Hi @SaitejaUtpala, this seems like it is a WIP. Will you complete the checklist - feel free to @ me afterwards

Yes @KennethEnevoldsen !

**Comment by bp-high** [comment]
Hi @KennethEnevoldsen had a doubt, this dataset was part of the semrel shared task hosted this year at semeval (https://semantic-textual-relatedness.github.io/) the authors released this dataset by re-annotating collected data in the different languages using the respective native speakers and best-worst scaling methodology. (https://arxiv.org/pdf/2402.08638.pdf , https://arxiv.org/pdf/2403.18933.pdf)

They try to differentiate this task from STS by providing the below definition and example:

> Two sentences are considered semantically similar when they have a paraphrasal or entailment relation. On the other hand, relatedness is a much broader concept that accounts for all the commonalities between two sentences: whether they are on the same topic, express the same view, originate from the same time period, one elaborates on (or follows from) the other, etc. For instance, for the following sentence pairs:
> PAIRS   SENTENCE 1                                                    SENTENCE 2
1              There was a lemon tree next to the house.   The boy enjoyed reading under the lemon tree.
2              There was a lemon tree next to the house.   The boy was an excellent football player.
> 
> 
> Most people will agree that the sentences in pair 1 are more related than the sentences in pair 2.

Given that they have given so much emphasis on STR being a much broader task than STS can this be considered a separate task for mteb/mmteb rather than being under the STS task group? Just wanted to know the opinion of the maintainers and other people in the field.

**Comment by KennethEnevoldsen** [comment]
Thanks for the comment @bp-high. Seems like the task is evaluated in the same way though (I only briefly skimmed the paper) so I would probably add the STR task subtype here.

**Comment by SaitejaUtpala** [comment]
@KennethEnevoldsen  This is done. Please take look into this.

File: mteb/tasks/STS/multilingual/STSSemRel2024.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        },
+        description=("Semantic Textual Relatedness for African and Asian Languages"),
+        reference="https://semantic-textual-relatedness.github.io/",
+        type="STS",
+        category="s2s",
+        eval_splits=_SPLITS,
+        eval_langs=_LANGUAGES,
+        main_score="cosine_spearman",
+        date=None,
+        form=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/STS/multilingual/STSSemRel2024.py:46]:
```suggestion
        date=None, # should be a date range when the dataset text was written (from, to)
        form=["written"], 
```

**Comment by SaitejaUtpala** [comment]
@KennethEnevoldsen  Resoloved and added points . We are  having a total of 13 datasets which includes 8 new datasets in STS ("afr", "arq", "amh", "hau", "ind", "kin", "arb", "ary" ). 13*2 + 8*4 =  58 points.

**Comment by SaitejaUtpala** [comment]
@KennethEnevoldsen  task_subtypes=["Semantic Text Relatedness"], this is causing some errors, probably we have to define this somewhere. Let me know where ?

**Comment by KennethEnevoldsen** [comment]
@SaitejaUtpala I noticed a few more issues here as well, the tests also fail (you should add STR to task types)

**Comment by KennethEnevoldsen** [comment]
@SaitejaUtpala it seems like this PR hasn't been worked on for a while. I will close it for now, but do feel free to re-open it.

**Pull request closed without merging**